### PR TITLE
Add absolute humidity component

### PR DESCRIPTION
**Pull request by DAVe3283**
## Description:

Adds a component to calculate the absolute humidity provided air temperature and relative humidity from other sensors.

**Related issue (if applicable):** N/A

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#4519

## Checklist:

  - [X] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [X] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/sensor/absolute_humidity.rst [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      # Use any temperature and relative humidity source, e.g. a BME280:
+      - platform: bme280
+        temperature:
+          name: Temperature
+          id: air_temperature
+        pressure:
+          name: Air Pressure
+        humidity:
+          name: Relative Humidity
+          id: relative_humidity
```
Comment (jesserockz) [inline components/sensor/absolute_humidity.rst]: Lets just leave this as a blank template as people will copy paste this and it wont match their setup
```suggestion
      - platform: ...
        temperature:
          name: Temperature
          id: air_temperature
        humidity:
          name: Relative Humidity
          id: relative_humidity
```

**Review by jesserockz — CHANGES_REQUESTED** [review]
Im fine with you adding the link to the different sensors, but not sure about deleting all the references between sensors. Lets leave them there in this PR and we can visit removing/tidying in a new PR if need be...
Same with altering the scd4x example yaml.
I am fine with updating the bme280 example to use this sensor.

**Comment by DAVe3283** [comment]
I had noticed it was very inconsistent what sensors linked to what other sensors, and it seemed that was all a holdover from when there were only a handful of sensors supported. I agree, that is out of scope for this PR though. Once this is all merged in, I can create separate PRs to clean up the issues I noticed in the sensor components.

**Comment by jesserockz** [comment]
The reason it is inconsistent is because people don't go back and add them after new ones are added. It would actually be better to link to the homepage category for the similar sensor type tbh.

**Review by jesserockz — APPROVED** [review]

**Pull request merged**